examples/warm_up_example/fedavg_warm_up/client.py

```diff
@@ -0,0 +1,90 @@
+import argparse
+from logging import INFO
+from pathlib import Path
+
+import flwr as fl
+import torch
+import torch.nn as nn
+from flwr.common.logger import log
+from flwr.common.typing import Config, Tuple
+from torch.nn.modules.loss import _Loss
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+
+from examples.models.cnn_model import MnistNet
+from fl4health.checkpointing.checkpointer import LatestTorchCheckpointer
+from fl4health.clients.basic_client import BasicClient
+from fl4health.utils.load_data import load_mnist_data
+from fl4health.utils.metrics import Accuracy
+from fl4health.utils.random import set_all_random_seeds
+from fl4health.utils.sampler import DirichletLabelBasedSampler
+
+
+class MnistFedAvgClient(BasicClient):
+    def get_data_loaders(self, config: Config) -> Tuple[DataLoader, DataLoader]:
+        sampler = DirichletLabelBasedSampler(list(range(10)), sample_percentage=0.75, beta=1)
+        batch_size = self.narrow_config_type(config, "batch_size", int)
+        train_loader, val_loader, _ = load_mnist_data(self.data_path, batch_size, sampler)
+        return train_loader, val_loader
+
+    def get_model(self, config: Config) -> nn.Module:
+        return MnistNet().to(self.device)
+
+    def get_optimizer(self, config: Config) -> Optimizer:
+        return torch.optim.AdamW(self.model.parameters(), lr=0.01)
+
+    def get_criterion(self, config: Config) -> _Loss:
+        return torch.nn.CrossEntropyLoss()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="FL Client Main")
+    parser.add_argument("--dataset_path", action="store", type=str, help="Path to the local dataset")
+    parser.add_argument(
+        "--server_address",
+        action="store",
+        type=str,
+        help="Server Address for the clients to communicate with the server through",
+        default="0.0.0.0:8080",
+    )
+    parser.add_argument(
+        "--seed",
+        action="store",
+        type=int,
+        help="Seed for the random number generator",
+        required=False,
+    )
+    parser.add_argument(
+        "--checkpoint_dir",
+        action="store",
+        type=str,
+        help="Path to the directory where the checkpoints are stored",
+        required=True,
+    )
+    parser.add_argument(
```

Comment:
I don't mind having this argument. However, if you're setting the random seed, each client actually generates a unique hash when it starts up under the property `self.client_name` using the `generate_hash` function. That should be deterministic with the random seed and you could eliminate this argument. I don't care either way, just wanted to mention it as a potential simplification.

Reply:
That's a good point. I was following flabby experiments for this argument, but I will update based on this comment and remove extra arguments.

Reply:
Yeah I see what you mean! The reason the FLamby experiments use it is because the clients numbers specifically correspond to the data loaders. So we explicitly needed to set the numbers. Again, no worries if you want to leave it though.